TST: `test_expm_multiply_dtype`  failure on aarch64 (flaky test)

### Describe your issue.

test_expm_multiply_dtype[True-float32-float32]  failed in one build for aarch64 when building scipy for Fedora.
I saw this test being discussed here: https://github.com/scipy/scipy/issues/16386
And adjusted here: https://github.com/scipy/scipy/pull/16550

It seems like the test is still flaky.

edit: I run some other builds, and so far it failed only in 1 of 4 builds.
### Reproducing Code Example

```python
Build scipy on aarch64
```


### Error message

```shell
=================================== FAILURES ===================================
________________ test_expm_multiply_dtype[True-float32-float32] ________________
[gw5] linux -- Python 3.12.0 /usr/bin/python3
dtype_a = <class 'numpy.float32'>, dtype_b = <class 'numpy.float32'>
b_is_matrix = True
    @pytest.mark.parametrize("dtype_a", DTYPES)
    @pytest.mark.parametrize("dtype_b", DTYPES)
    @pytest.mark.parametrize("b_is_matrix", [False, True])
    def test_expm_multiply_dtype(dtype_a, dtype_b, b_is_matrix):
        """Make sure `expm_multiply` handles all numerical dtypes correctly."""
        assert_allclose_ = (partial(assert_allclose, rtol=1.2e-3, atol=1e-5)
                            if {dtype_a, dtype_b} & IMPRECISE else assert_allclose)
        rng = np.random.default_rng(1234)
        # test data
        n = 7
        b_shape = (n, 3) if b_is_matrix else (n, )
        if dtype_a in REAL_DTYPES:
            A = scipy.linalg.inv(rng.random([n, n])).astype(dtype_a)
        else:
            A = scipy.linalg.inv(
                rng.random([n, n]) + 1j*rng.random([n, n])
            ).astype(dtype_a)
        if dtype_b in REAL_DTYPES:
            B = (2*rng.random(b_shape)).astype(dtype_b)
        else:
            B = (rng.random(b_shape) + 1j*rng.random(b_shape)).astype(dtype_b)
    
        # single application
        sol_mat = expm_multiply(A, B)
        sol_op = estimated(expm_multiply)(aslinearoperator(A), B)
        direct_sol = np.dot(sp_expm(A), B)
        assert_allclose_(sol_mat, direct_sol)
>       assert_allclose_(sol_op, direct_sol)
scipy/sparse/linalg/tests/test_expm_multiply.py:333: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (<function assert_allclose.<locals>.compare at 0xffff9d4bf560>, array([[ 1.6330003 ,  0.9154305 ,  1.4547368 ],
      ...738 ],
       [ 2.6518312 ,  2.582695  ,  1.6677312 ],
       [ 0.9006202 ,  1.1473742 ,  2.0856612 ]], dtype=float32))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.0012, atol=1e-05', 'verbose': True}
    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Not equal to tolerance rtol=0.0012, atol=1e-05
E           
E           Mismatched elements: 1 / 21 (4.76%)
E           Max absolute difference: 0.00039673
E           Max relative difference: 0.00157336
E            x: array([[ 1.633   ,  0.91543 ,  1.454737],
E                  [ 1.856501,  1.413198,  2.325316],
E                  [-0.24091 ,  1.7787  ,  1.782193],...
E            y: array([[ 1.633085,  0.91549 ,  1.455039],
E                  [ 1.856446,  1.413219,  2.325555],
E                  [-0.24129 ,  1.778732,  1.78204 ],...
/usr/lib64/python3.12/contextlib.py:81: AssertionError
=============================== warnings summary ===============================
_lib/tests/test__util.py::test_pool
_lib/tests/test__util.py::test_pool
_lib/tests/test__util.py::test_mapwrapper_parallel
_lib/tests/test__util.py::test_mapwrapper_parallel
_lib/tests/test__util.py::test_mapwrapper_parallel
_lib/tests/test__util.py::test_mapwrapper_parallel
  /usr/lib64/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=32878) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
fft/tests/test_multithreading.py: 2 warnings
fft/tests/test_numpy.py: 8 warnings
  /usr/lib64/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=32881) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
_lib/tests/test_warnings.py: 89 warnings
  /builddir/build/BUILDROOT/scipy-1.11.1-1.fc39.aarch64/usr/lib64/python3.12/site-packages/scipy/_lib/tests/test_warnings.py:42: DeprecationWarning: Attribute s is deprecated and will be removed in Python 3.14; use value instead
    if node.args[0].s == "ignore":
integrate/tests/test__quad_vec.py: 24 warnings
optimize/tests/test__differential_evolution.py: 4 warnings
optimize/tests/test__shgo.py: 4 warnings
  /usr/lib64/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=32887) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
optimize/tests/test_optimize.py::TestBrute::test_workers
optimize/tests/test_optimize.py::TestBrute::test_workers
  /usr/lib64/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=32900) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
stats/tests/test_stats.py::TestMGCStat::test_workers
stats/tests/test_stats.py::TestMGCStat::test_workers
  /usr/lib64/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=32884) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED scipy/sparse/linalg/tests/test_expm_multiply.py::test_expm_multiply_dtype[True-float32-float32]
= 1 failed, 54377 passed, 3022 skipped, 224 xfailed, 10 xpassed, 141 warnings in 244.55s (0:04:04)
```


### SciPy/NumPy/Python version and system information

```shell
scipy 1.11.1;
python3-numpy                  aarch64    1:1.24.3-2.fc39 
python3                        aarch64    3.12.0~b4-1.fc39
```

There was another failure reported in https://github.com/scipy/scipy/pull/20649#issuecomment-2103253789, and I just got one on gh-21496:

```
FAILED sparse/linalg/tests/test_expm_multiply.py::test_expm_multiply_dtype[True-float32-float32] - AssertionError: 
Not equal to tolerance rtol=0.0012, atol=1e-05

Mismatched elements: 1 / 21 (4.76%)
Max absolute difference: 0.00031149
Max relative difference: 0.00129095
 x: array([[ 1.632972,  0.915509,  1.455166],
       [ 1.856465,  1.413229,  2.325555],
       [-0.240978,  1.778699,  1.781919],...
 y: array([[ 1.633085,  0.91549 ,  1.455039],
       [ 1.856447,  1.413218,  2.325553],
       [-0.24129 ,  1.778732,  1.78204 ],...
```

I'll give the test tolerance another tweak.

--- a/scipy/sparse/linalg/tests/test_expm_multiply.py
+++ b/scipy/sparse/linalg/tests/test_expm_multiply.py
@@ -311,7 +311,7 @@ class TestExpmActionInterval:
 @pytest.mark.parametrize("b_is_matrix", [False, True])
 def test_expm_multiply_dtype(dtype_a, dtype_b, b_is_matrix):
     """Make sure `expm_multiply` handles all numerical dtypes correctly."""
-    assert_allclose_ = (partial(assert_allclose, rtol=1.2e-3, atol=1e-5)
+    assert_allclose_ = (partial(assert_allclose, rtol=1.8e-3, atol=1e-5)
                         if {dtype_a, dtype_b} & IMPRECISE else assert_allclose)
     rng = np.random.default_rng(1234)
     # test data

